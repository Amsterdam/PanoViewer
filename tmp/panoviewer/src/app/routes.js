import PAGES from './pages';

export const ROUTER_NAMESPACE = 'atlasRouter';

const MAIN_PATHS = {
  DATA: 'data',
  DATASETS: 'datasets',
  CONTENT: 'content'
};

export const routing = {
  home: {
    title: 'Home',
    path: '/',
    type: `${ROUTER_NAMESPACE}/${PAGES.HOME}`,
    page: PAGES.HOME
  },
  data: {
    title: 'Data',
    path: `/${MAIN_PATHS.DATA}/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.DATA}`,
    page: PAGES.DATA
  },
  datasets: {
    title: 'Datasets',
    path: `/${MAIN_PATHS.DATASETS}/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.DATASETS}`,
    page: PAGES.DATASETS
  },
  addresses: {
    title: 'Adressen',
    path: `/${MAIN_PATHS.DATA}/bag/adressen/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.ADDRESSES}`,
    page: PAGES.ADDRESSES
  },
  establishments: {
    title: 'Vestigingen',
    path: `/${MAIN_PATHS.DATA}/hr/vestigingen/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.ESTABLISHMENTS}`,
    page: PAGES.ESTABLISHMENTS
  },
  cadastralObjects: {
    title: 'Kadastrale objecten',
    path: `/${MAIN_PATHS.DATA}/brk/kadastrale-objecten/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.CADASTRAL_OBJECTS}`,
    page: PAGES.CADASTRAL_OBJECTS
  },
  searchDatasets: {
    title: 'Datasets zoekresultaten',
    path: `/${MAIN_PATHS.DATASETS}/zoek/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.SEARCH_DATASETS}`,
    page: PAGES.SEARCH_DATASETS
  },
  datasetDetail: {
    title: 'Dataset',
    path: `/${MAIN_PATHS.DATASETS}/:id/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.DATASET_DETAIL}`,
    page: PAGES.DATASET_DETAIL
  },
  dataQuerySearch: {
    title: 'Data zoekresultaten',
    path: `/${MAIN_PATHS.DATA}/zoek/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.DATA_QUERY_SEARCH}`,
    page: PAGES.DATA_QUERY_SEARCH
  },
  dataGeoSearch: {
    title: 'Data zoekresultaten op locatie',
    path: `/${MAIN_PATHS.DATA}/geozoek/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.DATA_GEO_SEARCH}`,
    page: PAGES.DATA_GEO_SEARCH
  },
  dataSearchCategory: {
    title: 'Data zoekresultaten',
    path: `/${MAIN_PATHS.DATA}/:category/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.DATA_SEARCH_CATEGORY}`,
    page: PAGES.DATA_SEARCH_CATEGORY
  },
  panorama: {
    title: 'Panoramabeeld',
    path: `/${MAIN_PATHS.DATA}/panorama/:id/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.PANORAMA}`,
    page: PAGES.PANORAMA
  },
  nieuws: {
    title: 'Nieuws',
    path: `/${MAIN_PATHS.CONTENT}/nieuws/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.NEWS}`,
    page: PAGES.NEWS
  },
  help: {
    title: 'Help',
    path: `/${MAIN_PATHS.CONTENT}/help/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.HELP}`,
    page: PAGES.HELP
  },
  proclaimer: {
    title: 'Proclaimer',
    path: `/${MAIN_PATHS.CONTENT}/proclaimer/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.PROCLAIMER}`,
    page: PAGES.PROCLAIMER
  },
  bediening: {
    title: 'Bediening',
    path: `/${MAIN_PATHS.CONTENT}/bediening/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.CONTROLS}`,
    page: PAGES.CONTROLS
  },
  gegevens: {
    title: 'Gegevens',
    path: `/${MAIN_PATHS.CONTENT}/gegevens/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.DATA_INFO}`,
    page: PAGES.DATA_INFO
  },
  apis: {
    title: 'Api\'s',
    path: `/${MAIN_PATHS.CONTENT}/apis/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.ABOUT_API}`,
    page: PAGES.ABOUT_API
  },
  actuality: {
    title: 'Actualiteit',
    path: `/${MAIN_PATHS.CONTENT}/actualiteit/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.ACTUALITY}`,
    page: PAGES.ACTUALITY
  },
  privacy_beveiliging: {
    title: 'Privacy en informatiebeveiliging',
    path: `/${MAIN_PATHS.CONTENT}/privacy-en-informatiebeveiliging/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.PRIVACY_SECURITY}`,
    page: PAGES.PRIVACY_SECURITY
  },
  beschikbaar_kwaliteit: {
    title: 'Beschikbaarheid en kwaliteit van de data',
    path: `/${MAIN_PATHS.CONTENT}/beschikbaarheid-en-kwaliteit-data/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.AVAILABILITY_QUALITY}`,
    page: PAGES.AVAILABILITY_QUALITY
  },
  beheer_werkwijze: {
    title: 'Technisch beheer en werkwijze',
    path: `/${MAIN_PATHS.CONTENT}/technisch-beheer-en-werkwijze/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.MANAGEMENT}`,
    page: PAGES.MANAGEMENT
  },
  statistieken: {
    title: 'Statistieken',
    path: `/${MAIN_PATHS.CONTENT}/statistieken/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.STATISTICS}`,
    page: PAGES.STATISTICS
  },
  inloggen: {
    title: 'Inloggen',
    path: `/${MAIN_PATHS.CONTENT}/inloggen/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.LOGIN}`,
    page: PAGES.LOGIN
  },
  verplaatst: {
    title: 'Pagina verplaatst',
    path: '/verplaatst/',
    type: `${ROUTER_NAMESPACE}/${PAGES.MOVED}`,
    page: PAGES.MOVED
  },
  niet_gevonden: {
    title: 'Pagina niet gevonden',
    path: '/niet-gevonden/',
    type: `${ROUTER_NAMESPACE}/${PAGES.NOT_FOUND}`,
    page: PAGES.NOT_FOUND
  },
  dataDetail: {
    title: 'Data detail',
    path: `/${MAIN_PATHS.DATA}/:type/:subtype/:id/`,
    type: `${ROUTER_NAMESPACE}/${PAGES.DATA_DETAIL}`,
    page: PAGES.DATA_DETAIL
  }
};

// e.g. { home: '/' }, to be used by redux-first-router/connectRoutes
const routes = Object.keys(routing).reduce((acc, key) => {
  acc[routing[key].type] = routing[key].path;
  return acc;
}, {});

export default routes;
