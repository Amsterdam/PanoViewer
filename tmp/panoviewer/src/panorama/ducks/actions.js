// Actions creators
import {
  CLOSE_PANORAMA,
  FETCH_PANORAMA_ERROR,
  FETCH_PANORAMA_HOTSPOT_REQUEST,
  FETCH_PANORAMA_REQUEST,
  FETCH_PANORAMA_REQUEST_EXTERNAL,
  FETCH_PANORAMA_SUCCESS,
  SET_PANORAMA_LOCATION,
  SET_PANORAMA_ORIENTATION,
  SET_PANORAMA_TAGS
} from './constants';

export const fetchPanoramaRequest = (payload) => ({
  type: FETCH_PANORAMA_REQUEST,
  payload,
  meta: {
    tracking: true
  }
});

export const fetchPanoramaHotspotRequest = (payload) => ({
  type: FETCH_PANORAMA_HOTSPOT_REQUEST,
  payload,
  meta: {
    tracking: true
  }
});

export const setPanoramaTags = (payload) => ({
  type: SET_PANORAMA_TAGS,
  payload,
  meta: {
    tracking: payload
  }
});

export const fetchPanoramaRequestExternal = () => ({
  type: FETCH_PANORAMA_REQUEST_EXTERNAL,
  meta: {
    tracking: true
  }
});

export const fetchPanoramaSuccess = (payload) => ({
  type: FETCH_PANORAMA_SUCCESS,
  payload,
  meta: {
    tracking: payload
  }
});
export const fetchPanoramaError = (error) => ({
  type: FETCH_PANORAMA_ERROR,
  payload: error
});

export const closePanorama = () => ({
  type: CLOSE_PANORAMA
});
export const setPanoramaLocation = (payload) => ({
  type: SET_PANORAMA_LOCATION,
  payload,
  meta: {
    tracking: true
  }
});
export const setPanoramaOrientation = ({ heading, pitch, fov }) => ({
  type: SET_PANORAMA_ORIENTATION,
  payload: {
    heading,
    pitch,
    fov
  }
});
