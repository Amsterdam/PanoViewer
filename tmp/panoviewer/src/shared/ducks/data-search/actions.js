import {
  FETCH_GEO_SEARCH_RESULTS_FAILURE,
  FETCH_GEO_SEARCH_RESULTS_REQUEST,
  FETCH_GEO_SEARCH_RESULTS_SUCCESS_LIST,
  FETCH_GEO_SEARCH_RESULTS_SUCCESS_PANEL,
  FETCH_QUERY_SEARCH_MORE_RESULTS_REQUEST,
  FETCH_QUERY_SEARCH_MORE_RESULTS_SUCCESS,
  FETCH_QUERY_SEARCH_RESULTS_REQUEST,
  FETCH_QUERY_SEARCH_RESULTS_SUCCESS,
  REQUEST_NEAREST_DETAILS
} from './constants';

// Action creators
export const fetchMapSearchResultsRequest = (payload, isMap) => ({
  type: FETCH_GEO_SEARCH_RESULTS_REQUEST,
  payload,
  meta: {
    isMap
  }
});
export const fetchMapSearchResultsSuccessPanel = (results, numberOfResults) => ({
  type: FETCH_GEO_SEARCH_RESULTS_SUCCESS_PANEL,
  payload: { results, numberOfResults }
});

export const fetchMapSearchResultsSuccessList = (results, numberOfResults) => ({
  type: FETCH_GEO_SEARCH_RESULTS_SUCCESS_LIST,
  payload: { results, numberOfResults }
});

export const fetchMapSearchResultsFailure = (payload) => ({
  type: FETCH_GEO_SEARCH_RESULTS_FAILURE,
  payload
});

export const showSearchResults = (results, query, numberOfResults) => ({
  type: FETCH_QUERY_SEARCH_RESULTS_SUCCESS,
  payload: { results, numberOfResults },
  meta: {
    tracking: {
      query,
      numberOfResults
    }
  }
});

export const fetchSearchResultsByQuery = (payload, loadMore = false) => ({
  type: FETCH_QUERY_SEARCH_RESULTS_REQUEST,
  payload,
  meta: {
    loadMore
  }
});

export const fetchMoreResults = () => ({
  type: FETCH_QUERY_SEARCH_MORE_RESULTS_REQUEST
});

export const fetchMoreResultsSuccess = (payload) => ({
  type: FETCH_QUERY_SEARCH_MORE_RESULTS_SUCCESS,
  payload
});

export const requestNearestDetails = (payload) => ({
  type: REQUEST_NEAREST_DETAILS,
  payload
});
