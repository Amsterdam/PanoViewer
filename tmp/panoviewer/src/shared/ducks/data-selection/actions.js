import {
  CANCEL_DATA_SELECTION,
  END_DATA_SELECTION,
  FETCH_DATA_SELECTION_FAILURE,
  FETCH_DATA_SELECTION_REQUEST,
  FETCH_DATA_SELECTION_SUCCESS,
  DOWNLOAD_DATA_SELECTION,
  RESET_DATA_SELECTION,
  REMOVE_GEOMETRY_FILTER,
  SET_DATASET,
  SET_GEOMETRY_FILTER,
  SET_PAGE,
  START_DATA_SELECTION, FETCH_MARKERS_REQUEST, FETCH_MARKERS_SUCCESS, FETCH_MARKERS_FAILURE
} from './constants';

export const fetchDataSelection = (payload) => ({ type: FETCH_DATA_SELECTION_REQUEST, payload });
export const fetchMarkersRequest = () => ({ type: FETCH_MARKERS_REQUEST });
export const fetchMarkersSuccess = (payload) => ({ type: FETCH_MARKERS_SUCCESS, payload });
export const fetchMarkersFailure = (payload) => (
  { type: FETCH_MARKERS_FAILURE, payload, error: true }
);

export const setPage = (payload) => ({ type: SET_PAGE, payload });
export const setDataset = (payload) => ({ type: SET_DATASET, payload });
export const removeGeometryFilter = () => ({ type: REMOVE_GEOMETRY_FILTER });
export const receiveDataSelectionSuccess = (payload) => ({
  type: FETCH_DATA_SELECTION_SUCCESS,
  payload
});
export const receiveDataSelectionFailure = (payload) => ({
  type: FETCH_DATA_SELECTION_FAILURE,
  payload
});
export const downloadDataSelection = (payload) => ({
  type: DOWNLOAD_DATA_SELECTION,
  meta: {
    tracking: payload
  }
});
export const setGeometryFilter = (payload) => ({
  type: SET_GEOMETRY_FILTER,
  payload,
  meta: {
    tracking: true
  }
});
export const resetDrawing = (payload = false) => ({
  type: RESET_DATA_SELECTION,
  payload
});
export const cancelDrawing = () => ({
  type: CANCEL_DATA_SELECTION
});
export const endDataSelection = () => ({
  type: END_DATA_SELECTION,
  meta: {
    tracking: true
  }
});
export const startDrawing = () => ({
  type: START_DATA_SELECTION
});
