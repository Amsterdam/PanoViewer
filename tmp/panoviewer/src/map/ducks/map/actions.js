import { getLabelObjectByTags } from '../../../panorama/ducks/selectors';
import { normalizeCoordinate } from '../../../shared/services/coordinate-reference-system';
import {
  MAP_EMPTY_GEOMETRY, MAP_UPDATE_SHAPE, MAP_SET_DRAWING_MODE, MAP_END_DRAWING,
  MAP_CLEAR, MAP_ZOOM, TOGGLE_MAP_PANEL, CLOSE_MAP_PANEL, SET_MAP_BASE_LAYER,
  TOGGLE_MAP_OVERLAY, TOGGLE_MAP_OVERLAY_PANORAMA, TOGGLE_MAP_OVERLAY_VISIBILITY,
  MAP_PAN, SET_MAP_CLICK_LOCATION, MAP_BOUNDING_BOX, MAP_LOADING
} from './constants';

// Actions
export const mapEmptyGeometry = () => ({ type: MAP_EMPTY_GEOMETRY });
export const mapUpdateShape = (payload) => ({ type: MAP_UPDATE_SHAPE, payload });
export const mapSetDrawingMode = (payload) => ({
  type: MAP_SET_DRAWING_MODE,
  payload,
  meta: {
    tracking: payload.drawingMode
  }
});
export const mapEndDrawing = (payload) => ({ type: MAP_END_DRAWING, payload });
export const mapClear = () => ({ type: MAP_CLEAR });
export const updateZoom = (payload) => ({ type: MAP_ZOOM, payload });
export const toggleMapPanel = () => ({ type: TOGGLE_MAP_PANEL });
export const closeMapPanel = () => ({ type: CLOSE_MAP_PANEL });
export const setMapBaseLayer = (payload) => ({
  type: SET_MAP_BASE_LAYER,
  payload,
  meta: {
    tracking: payload
  }
});
export const toggleMapOverlay = (payload) => ({
  type: TOGGLE_MAP_OVERLAY,
  payload: {
    mapLayers: (payload.id) ? [payload.id] : payload.legendItems.map((overlay) => overlay.id)
  },
  meta: {
    tracking: payload
  }
});

export const toggleMapOverlayPanorama = (tags) => ({
  type: TOGGLE_MAP_OVERLAY_PANORAMA,
  payload: getLabelObjectByTags(tags).layerId
});

export const toggleMapOverlayVisibility = (mapLayerId, isVisible) => ({
  type: TOGGLE_MAP_OVERLAY_VISIBILITY,
  mapLayerId,
  isVisible: !isVisible
});
export const updatePan = (payload) => ({
  type: MAP_PAN,
  payload: {
    latitude: normalizeCoordinate(payload.lat, 7),
    longitude: normalizeCoordinate(payload.lng, 7)
  }
});
export const setSelectedLocation = (payload) => ({
  type: SET_MAP_CLICK_LOCATION,
  payload: {
    location: {
      latitude: normalizeCoordinate(payload.latlng.lat, 7),
      longitude: normalizeCoordinate(payload.latlng.lng, 7)
    }
  },
  meta: {
    tracking: true
  }
});
export const updateBoundingBox = (payload) => ({
  type: MAP_BOUNDING_BOX,
  payload
});

export const mapLoadingAction = (payload) => ({ type: MAP_LOADING, payload });
