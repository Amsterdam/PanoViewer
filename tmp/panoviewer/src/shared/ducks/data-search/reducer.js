import {
  FETCH_GEO_SEARCH_RESULTS_FAILURE,
  FETCH_GEO_SEARCH_RESULTS_REQUEST,
  FETCH_GEO_SEARCH_RESULTS_SUCCESS_LIST,
  FETCH_GEO_SEARCH_RESULTS_SUCCESS_PANEL,
  FETCH_QUERY_SEARCH_MORE_RESULTS_SUCCESS,
  FETCH_QUERY_SEARCH_RESULTS_FAILURE,
  FETCH_QUERY_SEARCH_RESULTS_REQUEST,
  FETCH_QUERY_SEARCH_RESULTS_SUCCESS,
  initialState,
  REDUCER_KEY,
  SET_QUERY_CATEGORY
} from './constants';
import { routing } from '../../../app/routes';
import { FETCH_DATA_SELECTION_REQUEST } from '../data-selection/constants';
import paramsRegistry from '../../../store/params-registry';

export { REDUCER_KEY as DATA_SEARCH_REDUCER };

export default function reducer(state = initialState, action) {
  const enrichedState = {
    ...state,
    ...paramsRegistry.getStateFromQueries(REDUCER_KEY, action)
  };

  switch (action.type) {
    case routing.dataSearchCategory.type:
      return {
        ...enrichedState,
        category: action.payload.category
      };

    case FETCH_QUERY_SEARCH_RESULTS_REQUEST:
      return {
        ...enrichedState,
        isLoading: true,
        resultsQuery: [],
        numberOfResults: 0,
        query: action.payload
      };

    case FETCH_QUERY_SEARCH_RESULTS_SUCCESS: {
      const { results, numberOfResults } = action.payload;
      return {
        ...enrichedState,
        isLoading: false,
        numberOfResults,
        resultsQuery: results
      };
    }

    case FETCH_QUERY_SEARCH_MORE_RESULTS_SUCCESS: {
      return {
        ...enrichedState,
        isLoading: false,
        numberOfResults: 999,
        resultsQuery: [{
          ...state.resultsQuery[0],
          ...action.payload[0]
        }]
      };
    }

    case FETCH_GEO_SEARCH_RESULTS_REQUEST:
      return {
        ...enrichedState,
        isLoading: true,
        geoSearch: action.payload
      };

    case FETCH_GEO_SEARCH_RESULTS_SUCCESS_PANEL: {
      const { results, numberOfResults } = action.payload;
      return {
        ...enrichedState,
        isLoading: false,
        numberOfResults,
        resultsMapPanel: results
      };
    }

    case FETCH_GEO_SEARCH_RESULTS_SUCCESS_LIST: {
      const { results, numberOfResults } = action.payload;
      return {
        ...enrichedState,
        isLoading: false,
        numberOfResults,
        resultsMap: results
      };
    }

    case FETCH_QUERY_SEARCH_RESULTS_FAILURE:
    case FETCH_GEO_SEARCH_RESULTS_FAILURE:
      return {
        ...enrichedState,
        isLoading: false,
        error: action.payload
      };

    case SET_QUERY_CATEGORY:
      return {
        ...enrichedState,
        category: action.payload
      };

    case FETCH_DATA_SELECTION_REQUEST:
      return {
        ...enrichedState
      };

    default:
      return enrichedState;
  }
}
