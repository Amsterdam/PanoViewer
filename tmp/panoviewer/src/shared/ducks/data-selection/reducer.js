import PAGES from '../../../app/pages';
import { shouldResetState } from '../../../store/redux-first-router/actions';
import {
  FETCH_DATA_SELECTION_FAILURE,
  FETCH_DATA_SELECTION_REQUEST,
  FETCH_DATA_SELECTION_SUCCESS,
  FETCH_MARKERS_FAILURE,
  FETCH_MARKERS_REQUEST,
  FETCH_MARKERS_SUCCESS,
  initialState,
  REDUCER_KEY,
  REMOVE_GEOMETRY_FILTER,
  RESET_DATA_SELECTION,
  ROUTE_DATASET_MAPPER,
  SET_DATASET,
  SET_GEOMETRY_FILTER,
  SET_PAGE
} from './constants';
import { routing } from '../../../app/routes';
import { SET_SELECTION } from '../selection/selection';
import { FETCH_MAP_DETAIL_SUCCESS } from '../../../map/ducks/detail/constants';
import paramsRegistry from '../../../store/params-registry';

export { REDUCER_KEY as DATA_SELECTION };

export default function reducer(state = initialState, action) {
  if (shouldResetState(action, [PAGES.ADDRESSES, PAGES.ESTABLISHMENTS, PAGES.CADASTRAL_OBJECTS])) {
    return initialState;
  }
  const enrichedState = {
    ...state,
    ...paramsRegistry.getStateFromQueries(REDUCER_KEY, action),
    ...(ROUTE_DATASET_MAPPER[action.type]) ? { dataset: ROUTE_DATASET_MAPPER[action.type] } : {}
  };

  switch (action.type) {
    case routing.home.type:
    case FETCH_MAP_DETAIL_SUCCESS:
    case SET_SELECTION: {
      return {
        ...initialState
      };
    }

    case FETCH_MARKERS_REQUEST:
      return {
        ...enrichedState,
        loadingMarkers: true
      };

    case FETCH_DATA_SELECTION_REQUEST:
      return {
        ...enrichedState,
        dataset: action.payload.dataset,
        page: action.payload.page,
        isLoading: true
      };

    case FETCH_DATA_SELECTION_SUCCESS: {
      return {
        ...enrichedState,
        isLoading: false,
        markers: [],
        errorMessage: '',
        authError: false,
        ...action.payload
      };
    }

    case FETCH_DATA_SELECTION_FAILURE:
      return {
        ...enrichedState,
        isLoading: false,
        authError: (action.payload === 'Unauthorized'),
        errorMessage: action.payload,
        result: {},
        markers: []
      };

    case FETCH_MARKERS_FAILURE:
      return {
        ...enrichedState,
        loadingMarkers: false,
        errorMessage: action.payload,
        result: {},
        markers: []
      };

    case FETCH_MARKERS_SUCCESS:
      return {
        ...enrichedState,
        loadingMarkers: false,
        markers: action.payload
      };

    case SET_DATASET:
      return {
        ...enrichedState,
        dataset: action.payload
      };

    case SET_GEOMETRY_FILTER: {
      const { markers, distanceTxt, areaTxt } = action.payload;
      return {
        ...enrichedState,
        geometryFilter: {
          markers,
          description: `${distanceTxt} en ${areaTxt}`
        }
      };
    }

    case REMOVE_GEOMETRY_FILTER:
      return {
        ...enrichedState,
        geometryFilter: {}
      };

    case SET_PAGE:
      return {
        ...enrichedState,
        page: action.payload
      };

    case RESET_DATA_SELECTION:
      return {
        ...enrichedState,
        shape: ''
      };

    default:
      return enrichedState;
  }
}
