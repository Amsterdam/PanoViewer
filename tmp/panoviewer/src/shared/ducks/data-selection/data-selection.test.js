import reducer from './reducer';
import * as actionCreators from './actions';
import { routing } from '../../../app/routes';

describe('Data Selection Reducer', () => {
  /**
   * Use this helper to build an object that we can iterate the tests with.
   * @param actionCreatorName, this should be the name of the actionCreator function,
   * derived from the actionCreators you imported. use actionCreators[actionCreator].name to bind
   * the actionCreator to it's function name
   * @param expectedKeysToChange, used to pass the initial reducer state to test
   * @param [payload]: this must be an array, as action creators could accept more arguments.
   * @param [initialState], used to pass the initial reducer state to test, eg. if we conditionally
   * change a value of a state in the reducer.
   * @returns {{}}
   */
  const getExpectations = (
    actionCreatorName,
    expectedKeysToChange,
    payload = [],
    initialState = {}
  ) => ({
    [actionCreatorName]: {
      expectedKeysToChange,
      payload,
      initialState
    }
  });

  // Create the expectations what the actions would do here
  const expectations = {
    ...getExpectations(
      actionCreators.fetchDataSelection.name,
      ['isLoading', 'page', 'dataset'],
      ['dataset']
    ),
    ...getExpectations(
      actionCreators.setPage.name,
      ['page'],
      [1]
    ),
    ...getExpectations(
      actionCreators.setDataset.name,
      ['dataset'],
      ['foobar']
    ),
    ...getExpectations(
      actionCreators.fetchMarkersRequest.name,
      ['loadingMarkers'],
      []
    ),
    ...getExpectations(
      actionCreators.fetchMarkersSuccess.name,
      ['loadingMarkers', 'markers'],
      ['foobar']
    ),
    ...getExpectations(
      actionCreators.fetchMarkersFailure.name,
      ['loadingMarkers', 'errorMessage', 'result', 'markers'],
      ['error']
    ),
    ...getExpectations(
      actionCreators.setGeometryFilter.name,
      ['geometryFilter'],
      [[{ filter: 'foo' }]]
    ),
    ...getExpectations(
      actionCreators.removeGeometryFilter.name,
      ['geometryFilter']
    ),
    ...getExpectations(
      actionCreators.startDrawing.name,
      ['']
    ),
    ...getExpectations(
      actionCreators.endDataSelection.name,
      ['']
    ),
    ...getExpectations(
      actionCreators.cancelDrawing.name,
      ['']
    ),
    ...getExpectations(
      actionCreators.resetDrawing.name,
      ['shape']
    ),
    ...getExpectations(
      actionCreators.receiveDataSelectionSuccess.name,
      ['isLoading', 'markers', 'errorMessage', 'authError', 'data'],
      [{ data: { some: 'data' } }]
    ),
    ...getExpectations(
      actionCreators.receiveDataSelectionFailure.name,
      ['isLoading', 'authError', 'errorMessage', 'result', 'markers'],
      [{ error: 'error message' }]
    ),
    ...getExpectations(
      actionCreators.downloadDataSelection.name,
      [],
      []
    )
  };

  Object.keys(actionCreators).forEach((actionCreator) => {
    const { payload, expectedKeysToChange, initialState = {} } = expectations[actionCreator];
    it(`should set ${expectedKeysToChange.join(', ')} state when dispatching ${actionCreator}`, () => {
      const action = actionCreators[actionCreator](...payload);
      const result = reducer(initialState, action);
      expect(result).toMatchSnapshot();

      // Check if every key is changed, not more or less than the expected keys to change
      expect(expectedKeysToChange.sort().toString()).toEqual(Object.keys(result).sort().toString());
    });
  });

  describe('when a route type is dispatched', () => {
    it('should set dataset and an object from getStateFromQuery if meta.query is set', () => {
      const expectedKeysToChange = ['dataset'];
      const result = reducer({}, {
        type: routing.addresses.type,
        meta: {
          query: {}
        }
      });
      expect(result).toMatchSnapshot();
      expect(expectedKeysToChange.sort().toString()).toEqual(Object.keys(result).sort().toString());
    });

    it('should set the dataset and view if meta.query is not set', () => {
      const expectedKeysToChange = ['dataset'];
      const result = reducer({}, {
        type: routing.establishments.type
      });
      expect(result).toMatchSnapshot();
      expect(expectedKeysToChange.sort().toString()).toEqual(Object.keys(result).sort().toString());
    });
  });
});
