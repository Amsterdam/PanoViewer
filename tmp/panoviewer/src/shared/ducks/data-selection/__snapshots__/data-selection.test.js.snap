// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Selection Reducer should set  state when dispatching cancelDrawing 1`] = `Object {}`;

exports[`Data Selection Reducer should set  state when dispatching downloadDataSelection 1`] = `Object {}`;

exports[`Data Selection Reducer should set  state when dispatching endDataSelection 1`] = `Object {}`;

exports[`Data Selection Reducer should set  state when dispatching startDrawing 1`] = `Object {}`;

exports[`Data Selection Reducer should set dataset state when dispatching setDataset 1`] = `
Object {
  "dataset": "foobar",
}
`;

exports[`Data Selection Reducer should set geometryFilter state when dispatching removeGeometryFilter 1`] = `
Object {
  "geometryFilter": Object {},
}
`;

exports[`Data Selection Reducer should set geometryFilter state when dispatching setGeometryFilter 1`] = `
Object {
  "geometryFilter": Object {
    "description": "undefined en undefined",
    "markers": undefined,
  },
}
`;

exports[`Data Selection Reducer should set isLoading, authError, errorMessage, result, markers state when dispatching receiveDataSelectionFailure 1`] = `
Object {
  "authError": false,
  "errorMessage": Object {
    "error": "error message",
  },
  "isLoading": false,
  "markers": Array [],
  "result": Object {},
}
`;

exports[`Data Selection Reducer should set isLoading, markers, errorMessage, authError, data state when dispatching receiveDataSelectionSuccess 1`] = `
Object {
  "authError": false,
  "data": Object {
    "some": "data",
  },
  "errorMessage": "",
  "isLoading": false,
  "markers": Array [],
}
`;

exports[`Data Selection Reducer should set isLoading, page, dataset state when dispatching fetchDataSelection 1`] = `
Object {
  "dataset": undefined,
  "isLoading": true,
  "page": undefined,
}
`;

exports[`Data Selection Reducer should set loadingMarkers state when dispatching fetchMarkersRequest 1`] = `
Object {
  "loadingMarkers": true,
}
`;

exports[`Data Selection Reducer should set loadingMarkers, errorMessage, result, markers state when dispatching fetchMarkersFailure 1`] = `
Object {
  "errorMessage": "error",
  "loadingMarkers": false,
  "markers": Array [],
  "result": Object {},
}
`;

exports[`Data Selection Reducer should set loadingMarkers, markers state when dispatching fetchMarkersSuccess 1`] = `
Object {
  "loadingMarkers": false,
  "markers": "foobar",
}
`;

exports[`Data Selection Reducer should set page state when dispatching setPage 1`] = `
Object {
  "page": 1,
}
`;

exports[`Data Selection Reducer should set shape state when dispatching resetDrawing 1`] = `
Object {
  "shape": "",
}
`;

exports[`Data Selection Reducer when a route type is dispatched should set dataset and an object from getStateFromQuery if meta.query is set 1`] = `
Object {
  "dataset": "bag",
}
`;

exports[`Data Selection Reducer when a route type is dispatched should set the dataset and view if meta.query is not set 1`] = `
Object {
  "dataset": "hr",
}
`;
